cartography/intel/crxcavator/crxcavator.py

```diff
@@ -134,7 +155,35 @@ def transform_extensions(extension_details):
             'price': webstore.get('price'),
             'report_link': f"https://crxcavator.io/report/{extension_id}/{version}",
         })
-    return extensions
+    return extensions, list(permissions_set), extensions_permissions
+
+
+def parse_permissions_dict(permissions_dict):
+    """
+    Parses the possible complex permissions objects into a list of objects
+    Parsing cases based on results from crxcavator API as of October 2019
+    :param permissions_dict: a dict object from the permissions key
+    :return: a list of individual permissions
+    """
+    permissions = []
+    usb_devices = permissions_dict.get('usbDevices', [])
+    for device in usb_devices:
+        permissions.append(f"usbdevice-{device.get('productId', 'none')}-{device.get('vendorId', 'none')}")
+    filesystem = permissions_dict.get('fileSystem', [])
+    for filesystem_permission in filesystem:
+        permissions.append(f"filesystem-{filesystem_permission}")
+    socket = permissions_dict.get('socket', [])
+    for socket_permission in socket:
+        permissions.append(f"socket-{socket_permission}")
+    media_galleries = permissions_dict.get('mediaGalleries', [])
+    for media in media_galleries:
+        permissions.append(f"mediagalleries-{media}")
+    if len(permissions) == 0:
+        # this is a case not currently handled, so return the raw json object for ingestion
+        permission = json.dumps(permissions_dict)
+        logger.warning(f"Unknown permissions dict type {permission}")
+        permissions.append(permission)
```

Comment:
Do we still want to add this if we don't know what the permission is?

`json.dumps()` yields a string representation of the dict and I think it'll include brackets which could make merging it into a neo4j node messy.

Reply:
removed this, logging only.